implementations/code_df/code_changes_lines.py

```diff
@@ -0,0 +1,128 @@
+from datetime import datetime
+
+import conditions
+import utils
+from commit import Commit
+
+
+class CodeChangesLines(Commit):
+    """
+    Class for Code_Changes_Lines
+
+    :param items: A list of dictionaries, each element a line from the
+        JSON file with Perceval data
+
+    :param date_range: A tuple which represents the start and end date of
+        interest
+
+    :param is_code:  list of CodeCondition objects
+        It is used to determine what comprises source code.
+
+    :param conds: A list of Commit objects. It is used to
+        exclude or include different kinds of commits, like
+        empty or merge commits, or commits made on the master
+        branch.
+    """
+
+    def __init__(self, items, date_range=(None, None),
+                 is_code=[conditions.Naive()], conds=[]):
+        super().__init__(items, date_range, is_code, conds)
+
+        self.df = self._add_lines_modified_cols(self.df)
+
+    def compute(self):
+        """
+        Count the number of lines added or deleted in the data fetched
+        by Perceval.
+
+        It computes the sums of the 'additions' and 'deletions' columns
+        in the DataFrame.
+
+        :returns modified: A tuple representing the number of lines
+            modified. modified -> (additions, deletions)
+        """
+
+        df = self.df
+        additions = df['additions'].sum()
+        deletions = df['deletions'].sum()
+
+        modified = (additions, deletions)
+
+        return modified
+
+    def _agg(self, df, period):
+        """
+        Perform an aggregation operation on a DataFrame or Series
+        to find the total number of lines modified in a every interval
+        of the period specified in the time_series method, like
+        'M', 'W',etc.
+
+        It adds the number of lines modified for every row in the
+        series.
+
+        :param df: a pandas DataFrame on which the aggregation will be
+            applied.
+
+        :param period: A string which can be any one of the pandas time
+            series rules:
+            'W': week
+            'M': month
+            'D': day
+
+        :returns df: The aggregated dataframe, where aggregations have
+            been perform on "additions" and "deletions" columns
+        """
+
+        df = df.resample(period).agg({"additions": 'sum', "deletions": 'sum'})
```

Comment:
Not sure this is working: changed = added + deleted...

Reply:
Will change it